### PR TITLE
gitit: init at 0.13.0.0

### PR DESCRIPTION
**Pull request by Profpatsch**
###### Motivation for this change

The new gitit upstream release has unbroken the tool.

The second commit makes it into a top-level semi-static binary with a reasonable default runtime closure. 140 MB oO

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Profpatsch** [comment]
cc @P-E-Meunier

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -1405,6 +1405,9 @@ self: super: {
   # $HOME, which we don't have in our build sandbox.
   cabal-install-parsers = dontCheck super.cabal-install-parsers;
 
+  # gitit is unbroken in the latest release
+  gitit = markUnbroken super.gitit;
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-common.nix:1409]: If you submit this PR to the `haskell-updates` branch, then you shouldn't need this.

As soon as the daily update CI thing runs, `gitit` will be unmarked broken.

Comment (Profpatsch) [reply pkgs/development/haskell-modules/configuration-common.nix:1409]: Is the workings of that written down somewhere?

Comment (cdepillabout) [reply pkgs/development/haskell-modules/configuration-common.nix:1409]: @Profpatsch Unfortunately, not as far as I know.

I wrote about it a little in this post:

https://discourse.nixos.org/t/haskellpackages-stm-containers-fails-to-build/5416/4

Also, @maralorn recently created a blog post that included a short section explaining how this works:

https://mpickering.github.io/ide/posts/2020-06-05-ghcide-and-nixpkgs.html

Comment (maralorn) [reply pkgs/development/haskell-modules/configuration-common.nix:1409]: I fear I didn't mention the fact that hackage2nix generates the broken flags explicitly.

Comment (Profpatsch) [reply pkgs/development/haskell-modules/configuration-common.nix:1409]: I am still none the wiser. Is there a step-by-step description of what I have to change where and commit to which branch in what order? 

I will merge this one here for now, but am open to doing it the “correct” way if somebody points me to clear documentation that is not spread over multiple Youtube videos and blogposts.

**Comment by Profpatsch** [comment]
@GrahamcOfBorg build gitit

File: pkgs/development/haskell-modules/configuration-common.nix (hunk 1 above)
Comment (maralorn) [reply pkgs/development/haskell-modules/configuration-common.nix:1409]: @Profpatsch I completely understand your frustration. But for this narrow use case it's actually really simple:

1. Confirm that package build succeeds, when broken flag is overriden.
2. Make a PR against the `haskell-updates` branch, where you remove the package from the array of broken packages in `configuration-hackage2nix.yaml`.
3. After merge wait for the daily `hackage-packages.nix` update, which removes the broken flag from the package and removes the flag which disables builds on hydra.
4. Every friday `haskell-updates` get's merged into master.

Bonus: You can do step 3. on your own in your merge request, but yeah, I think the documentation for that is a youtube video right now.

We really need to make sure all of this is easily discoverable documented.

**Comment by maralorn** [comment]
It remains to note that after this merge `gitit` will not be build by hydra.

**Comment by Profpatsch** [comment]
Why wouldn’t it?

**Comment by maralorn** [comment]
Because `haskellPackages.gitit` has the flag `hydraPlatforms = stdenv.lib.platforms.none;` set in it's definition in `hackage-packages.nix`.

**Comment by peti** [comment]
I just tried compiling `gitit` and it does not build:

~~~~
onfiguring gitit-0.13.0.0...
CallStack (from HasCallStack):
  die', called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:1022:20 in Cabal-3.0.1.0:Distribution.Simple.Configure                   
  configureFinalizedPackage, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:475:12 in Cabal-3.0.1.0:Distribution.Simple.Configure
  configure, called at libraries/Cabal/Cabal/Distribution/Simple.hs:625:20 in Cabal-3.0.1.0:Distribution.Simple                                   
  confHook, called at libraries/Cabal/Cabal/Distribution/Simple/UserHooks.hs:65:5 in Cabal-3.0.1.0:Distribution.Simple.UserHooks                  
  configureAction, called at libraries/Cabal/Cabal/Distribution/Simple.hs:180:19 in Cabal-3.0.1.0:Distribution.Simple                             
  defaultMainHelper, called at libraries/Cabal/Cabal/Distribution/Simple.hs:116:27 in Cabal-3.0.1.0:Distribution.Simple                           
  defaultMain, called at Setup.lhs:3:10 in main:Main
Setup: Encountered missing or private dependencies:
hoauth2 >=1.3.0 && <1.12

builder for '/nix/store/v1ncl30rcavlhcb1qwdczj5f469mbzw1-gitit-0.13.0.0.drv' failed with exit code 1                               
~~~

**Comment by peti** [comment]
I should mention that this error occurred on haskell-updates, not on master. We'll merge haskell-updates to master soon, though.

**Pull request merged**